extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -32,34 +31,115 @@ public void setNodeName(final TransactionManagerConfiguration transactions) {
-            field.setAccessible(true);
-            field.set(null, new QuarkusPropertiesFactory(properties));
-
-        } catch (Exception e) {
-            log.error("Could not override transaction properties factory", e);
-        }
-
-        defaultProperties = properties;
-    }
-
     public void setDefaultTimeout(TransactionManagerConfiguration transactions) {
         transactions.defaultTransactionTimeout.ifPresent(defaultTimeout -> {
             arjPropertyManager.getCoordinatorEnvironmentBean().setDefaultTimeout((int) defaultTimeout.getSeconds());
             TxControl.setDefaultTimeout((int) defaultTimeout.getSeconds());
         });
     }
 
-    public static Properties getDefaultProperties() {
-        return defaultProperties;
-    }
-
     public void disableTransactionStatusManager() {
         arjPropertyManager.getCoordinatorEnvironmentBean()
                 .setTransactionStatusManagerEnable(false);
     }
+
+    private void setXaResourceOrphanFilterClassNames(JTAEnvironmentBean jtaEnvironmentBean) {
+        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateJTAXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinationManagerXAResourceOrphanFilter"));
+    }
+
+    private void setRecoveryModuleClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryModuleClassNames(Arrays.asList(
+                "com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.txoj.recovery.TORecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateAtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.CommitMarkableResourceRecordRecoveryModule"));
+    }
+
+    private void setExpiryScannerClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setExpiryScannerClassNames(Collections.singletonList(
+                "com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner"));
+    }
+
+    private void setRecoveryPort(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryPort(4712);
+    }
+
+    private void setRecoveryListener(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryListener(true);
+    }
+
+    public void setConfig(TransactionManagerConfiguration transactions) {
+        Properties systemProperties;
+        final SecurityManager sm = System.getSecurityManager();
```

Comment:
by interest, is the `SecurityManger` considered to need to be handled in Quarkus code when it's in general one app per one JVM system?

Reply:
Okay I removed the security check.